src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/ApplicableGroupsType.php

```diff
@@ -125,18 +132,33 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
         ;
 
         if ($this->isMultiShopEnabled) {
-            $shops = array_merge([
-                $this->trans('All stores', 'Admin.Global') => 0,
-            ], $this->shopByIdChoiceProvider->getChoices());
             $builder->add('shop_id', ChoiceType::class, [
                 'label' => false,
                 'required' => false,
                 'placeholder' => false,
-                'choices' => $shops,
+                'choices' => $this->buildShopChoices(),
             ]);
         }
     }
 
+    /**
+     * @return array<string, int>
+     */
+    private function buildShopChoices(): array
+    {
+        $choices = [];
```

Comment:
All stores should be put before the list of shops I think (even if it's not the initial default choice) so you can init you array like this:

```suggestion
        $choices = [
            $this->trans('All stores', 'Admin.Global') => 0,
        ];
```